cpp/src/fil/infer.cu

```diff
@@ -384,43 +385,47 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
     } else {
       finalize_class_label(out, num_rows);
     }
+    __syncthreads();  // free votes[] for next row set
```

Comment:
even though it's unnecessary in case `blocks_per_sm == 0`, it doesn't affect timing even in the most sensitive cases for the above algo. This allows not to leak another implementation detail to `__global__ void infer_k`

Reply:
This is fine.